liquidprompt

```diff
@@ -2341,8 +2347,22 @@ _lp_wifi() {
 # Get the battery status in percent.
 case "$LP_OS" in
     Linux)
-    _lp_battery() {
-        (( LP_ENABLE_BATT )) || return 5
+    __lp_battery_sysfs() {
+        local power_supply
+        for power_supply in "${_LP_LINUX_POWERSUPPLY_PATH}/"*; do
+            if [[ -r "${power_supply}/type" && $(< "${power_supply}/type") == 'Battery' && \
+                -r "${power_supply}/present" && $(< "${power_supply}/present") == '1' && \
+                -r "${power_supply}/status" && -r "${power_supply}/capacity" && \
+                ! -f "${power_supply}/scope" ]]; then
+                lp_battery_status=$(<"${power_supply}/status")
+                lp_battery=$(<"${power_supply}/capacity")
```

Comment:
This one is on me for suggesting it to you this way, but this other form of reading from a file is much much faster, and is how the rest of liquidprompt does it:
```suggestion
                IFS= read -r lp_battery_status <"${power_supply}/status"
                IFS= read -r lp_battery <"${power_supply}/capacity"
```